### PR TITLE
[Rest Server] export container port in api

### PR DESCRIPTION
**Pull request by abuccts**

### CODE REVIEW

File: rest-server/src/models/job.js [hunk 1]
```diff
@@ -387,10 +387,18 @@ class Job {
...
+            if (portStr.length > 0) {
+              const port = portStr.split(':');
+              containerPorts[port[0]] = port[1];
+            }
+          }
           jobDetail.taskRoles[taskRole].taskStatuses.push({
             taskIndex: task.taskIndex,
             containerId: task.containerId,
             containerIp: task.containerIp,
+            containerPorts: containerPorts,
```
Comment (qinchen123) [inline rest-server/src/models/job.js]: >containerPorts: containerPorts, [](start = 12, length = 31)

This will work ?

Comment (qinchen123) [reply rest-server/src/models/job.js]: containerPorts: containerPorts,


---
In reply to: [183613436](https://github.com/Microsoft/pai/pull/515#discussion_r183613436) [](ancestors = 183613436)

Comment (abuccts) [reply rest-server/src/models/job.js]: Where's the problem? The first is a string and the second is a variable.

Comment (qinchen123) [reply rest-server/src/models/job.js]: use a different name ? the name is the same. it is too easy to confuse.

File: webportal/src/app/job/job-view/job-detail-table.component.ejs [hunk 2]
```diff
@@ -84,6 +84,7 @@
       <th>Task Index</th>
       <th>Container Name</th>
       <th>Container IP</th>
+      <th>Container Port</th>
```
Comment (hwuu) [inline webportal/src/app/job/job-view/job-detail-table.component.ejs]: Port --> Ports?

File: webportal/src/app/job/job-view/job-detail-table.component.ejs [hunk 3]
```diff
@@ -98,6 +99,11 @@
           <td><%= taskRoles[taskRole].taskStatuses[i].taskIndex %></td>
           <td><%= taskRoles[taskRole].taskStatuses[i].containerId %></td>
           <td><%= taskRoles[taskRole].taskStatuses[i].containerIp %></td>
+          <td>
+            <% for (let portLabel of Object.keys(taskRoles[taskRole].taskStatuses[i].containerPort)) { %>
+              <p><%= portLabel %>: <%= taskRoles[taskRole].taskStatuses[i].containerPort[portLabel] %></p>
```
Comment (hwuu) [inline webportal/src/app/job/job-view/job-detail-table.component.ejs]: Separate by comma?

Comment (hwuu) [inline webportal/src/app/job/job-view/job-detail-table.component.ejs]: Better keep the list in one line?

File: webportal/src/app/job/job-view/job-detail-table.component.ejs (hunk 2 above)
Comment (abuccts) [reply webportal/src/app/job/job-view/job-detail-table.component.ejs]: updated

File: webportal/src/app/job/job-view/job-detail-table.component.ejs (hunk 3 above)
Comment (abuccts) [reply webportal/src/app/job/job-view/job-detail-table.component.ejs]: It's too long. One port type per line makes it clearer.

File: rest-server/src/models/job.js [hunk 4]
```diff
@@ -387,10 +387,18 @@ class Job {
...
+            if (portStr.length > 0) {
+              const port = portStr.split(':');
+              containerPorts[port[0]] = port[1];
+            }
+          }
           jobDetail.taskRoles[taskRole].taskStatuses.push({
             taskIndex: task.taskIndex,
             containerId: task.containerId,
             containerIp: task.containerIp,
+            containerPorts,
```
Comment (sterowang) [inline rest-server/src/models/job.js]: >containerPorts [](start = 12, length = 14)

What will JSON looks like if we write like this way?

**Pull request merged**